R/autoplot.R

```diff
@@ -0,0 +1,156 @@
+#'
+#' @param object An `epi_df`
+#' @param ... <[`tidy-select`][dplyr_tidy_select]> One or more unquoted
+#'   expressions separated by commas. Variable names can be used as if they
+#'   were positions in the data frame, so expressions like `x:y` can
+#'   be used to select a range of variables.
+#' @param .color_by Which variables should determine the color(s) used to plot
+#'   lines. Options include:
+#'   * `all_keys` - the default uses the interaction of any key variables
+#'     including the `geo_value`
+#'   * `geo_value` - `geo_value` only
+#'   * `other_keys` - any available keys that are not `geo_value`
+#'   * `.response` - the numeric variables (same as the y-axis)
+#'   * `all` - uses the interaction of all keys and numeric variables
+#'   * `none` - no coloring aesthetic is applied
+#' @param .facet_by Similar to `.color_by` except that the default is to display
+#'   each numeric variable on a separate facet
+#' @param .base_color Lines will be shown with this color. For example, with a
+#'   single numeric variable and faceting by `geo_value`, all locations would
+#'   share the same color line.
+#' @param .max_facets Cut down of the number of facets displayed. Especially
+#'   useful for testing when there are many `geo_value`'s or keys.
+#'
+#' @return A ggplot object
+#' @export
+#'
+#' @examples
+#' autoplot(jhu_csse_daily_subset, cases, death_rate_7d_av)
+#' autoplot(jhu_csse_daily_subset, case_rate_7d_av, .facet_by = "geo_value")
+#' autoplot(jhu_csse_daily_subset, case_rate_7d_av,
+#'   .color_by = "none",
+#'   .facet_by = "geo_value"
+#' )
+autoplot.epi_df <- function(
+    object, ...,
+    .color_by = c("all_keys", "geo_value", "other_keys", ".response", "all", "none"),
+    .facet_by = c(".response", "other_keys", "all_keys", "geo_value", "all", "none"),
+    .base_color = "#3A448F",
+    .max_facets = Inf) {
+  .color_by <- match.arg(.color_by)
+  .facet_by <- match.arg(.facet_by)
+
+  arg_is_scalar(.max_facets)
+  if (is.finite(.max_facets)) arg_is_int(.max_facets)
+  arg_is_chr_scalar(.base_color)
+
+  key_cols <- key_colnames(object)
+  non_key_cols <- setdiff(names(object), key_cols)
+  geo_and_other_keys <- kill_time_value(key_cols)
+
+  # --- check for numeric variables
+  allowed <- purrr::map_lgl(object[non_key_cols], is.numeric)
+  if (length(allowed) == 0) {
+    cli::cli_abort("No numeric variables were available to plot automatically.")
+  }
+  vars <- tidyselect::eval_select(rlang::expr(c(...)), object)
+  if (rlang::is_empty(vars)) { # find them automatically if unspecified
+    vars <- tidyselect::eval_select(names(allowed)[1], object)
+    cli::cli_warn(
+      "Plot variable was unspecified. Automatically selecting {.var {names(allowed)[1]}}."
+    )
+  } else { # if variables were specified, ensure that they are numeric
+    ok <- names(vars) %in% names(allowed)
+    if (!any(ok)) {
+      cli::cli_abort(
+        "None of the requested variables {.var {names(vars)}} are numeric."
+      )
```

Comment:
issue: I can't get this error to trigger. I have an epi_df with no numeric columns (sex is included as a key), but it gets plotted anyway.

![test data](https://github.com/cmu-delphi/epiprocess/assets/42820733/2a747e90-0517-4675-afd3-1eca9696eb6f)

The `fake` field is plotted if I run either `autoplot(aa, fake)` or `autoplot(aa)`

Reply:
suggestion: Please add tests to check that warnings and errors in this function are raised appropriately.

Reply:
I fixed this bug, but I don't think I have the bandwidth to test all the branches. Perhaps someone else can give it a go?

Reply:
Sure, I can do that.